internal/dag/gatewayapi_processor.go

```diff
@@ -540,7 +540,7 @@ func (p *GatewayAPIProcessor) computeTLSRoute(route *gatewayapi_v1alpha2.TLSRout
 		}
 
 		for host := range hosts {
-			secure := p.dag.EnsureSecureVirtualHost(ListenerName{Name: host, ListenerName: "ingress_https"})
```

Comment:
i vaguely remember this whole ListenerName thing was to enable more than just the 2 basic listeners

makes sense maybe to go back to this and could use a bigger refactor in the future once we support that? wdyt?

Reply:
Let me look at this some more, but one thought here is that I'd like to push more of the Listener logic into the DAG -- right now they're mostly built out in the xdscache logic itself, which feels off and inconsitent from other resource types. Perhaps as part of that refactor we can make sure that the DAG properly supports >2 listeners and that the xdscache can consume them.

Reply:
Yeah so I'm thinking that if we want to retain support for many Listeners, we should properly build that out in the DAG - model out all of the Listeners, attach vhosts to the relevant listener(s) there, and then this ListenerName field on the VirtualHost is probably not needed, we'll just traverse the association from Listener->VirtualHosts.

Reply:
I agree with that last comment, @skriss. I think that we definitely should be aiming to be able to do multiple listeners, so that in the future, we can allow some extra ports and have the managed Envoy stuff ensure the wiring.